templates/register2faOtp.html

```diff
@@ -29,17 +29,53 @@ <h4 class="section-title">2. Enter the 6 digits code </h4>
                         <input class='validate otp' type='number' name='otp' id='otp' max="999999" minlength="6" maxlength="6" autofocus/>
                         <label for='otp'>Use the entry starting with "<span name="data-connection-name"></span>"</label>
                         <span class="helper-text" data-error="Must be 6 digits" data-success="right"></span>
+                        <button type="button" class="waves-effect waves-light btn" id="otp-button">Submit</button>
                     </div>
                 </div>
                 <br/>
-                <div id="error" class="card-panel red accent-3 display-none" style="color: #ffffff;"><b>The code was invalid</b></div>
+                <div id="error" class="white card-panel red accent-3 display-none"><b>The code was invalid</b></div>
             </div>
         </div>
     </form>
     </center>
 
     <div class="section"></div>
 </main>
+<script nonce="{{.NonceCode}}">
+  const otpHandler = async function() {
+    const dataLength = $('#otp').val().length;
+    if(dataLength > 0) {
+        $("#error").hide();
+    }
+    if (dataLength == 6) {
+        await validateOneTimePass(false, $('#otp').val());
+        $('#otp').val("");
+    } else {
+      $("#error").show();
+    }
+  }
+  
+  const otcHander = async function() {
+    const dataLength = $('#otc').val().length;
+    if(dataLength > 0) {
+      $("#error").hide();
+    }
+    if (dataLength == 6) {
+        await validateOneTimePass(true, $('#otc').val());
+        $('#otc').val("");
+    } else {
+      $("#error").show();
+    }
+  }
+  
+  $('#otp-button').on('click', otpHandler);
+  
+  $('#otc-button').on('click', otcHander);
+  
+  document.getElementById('otp-button').addEventListener('touchstart', otpHandler);
+  
+  document.getElementById('otc-button').addEventListener('touchstart', otcHander);
+</script>
```

Comment:
Do you think this could be moved to templates/assets/script.js, in order to be with the rest of the Javascript stuff?

Reply:
Yes I will see if I can get that working.

Reply:
Moved